### PR TITLE
Use GitHub workflow for CI on PRs and master builds

### PR DESCRIPTION
**Pull request by srprash**
*Issue:* Our AppVeyor CI has been a hit or miss lately. Although it does [run](https://ci.appveyor.com/project/yogiraj07/aws-xray-sdk-dotnet-nd0el/builds/42783210), the CI is not able to post results back to Github PR. As a result the PRs are not blocked even if the AppVeyor CI fails. This is dangerous as we are flying blind. This is good point to use native GH workflows to run CI and we don't have to maintain AppVeyor CI anymore.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jon-armen** [comment]
I have several PR's where I'm waiting on a release.  Anything I may be able to assist with here to help get your builds running?

File: .github/workflows/ci.yml
Comment (NathanielRN) [inline .github/workflows/ci.yml:21]:
```suggestion
                version: [net452, netcoreapp2.0, netcoreapp3.1]
```

File: .github/workflows/ci.yml
Comment (NathanielRN) [inline .github/workflows/ci.yml:42]:
```suggestion
          run: >-
            dotnet test sdk/test/UnitTests/bin/Release/${{matrix.version}}/AWSXRayRecorder.UnitTests.dll
            --configuration Release
            --no-build
            --logger "console;verbosity=detailed"
```

**Review by NathanielRN — APPROVED** [review]
Looks good to me in principal. Do you have a test run where we can check it passes successfully?

**Review by jon-armen — APPROVED** [review]

**Comment by srprash** [comment]
@NathanielRN 
> Do you have a test run where we can check it passes successfully?
You can see the run in this PR itself or on my fork [here](https://github.com/srprash/aws-xray-sdk-dotnet/actions/runs/1931163210).

@jon-armen 
Thanks for offering. I think we are good to merge this now.

File: .github/workflows/ci.yml
Comment (srprash) [reply .github/workflows/ci.yml]: This is neat. Thanks!

File: .github/workflows/ci.yml
Comment (jon-armen) [inline .github/workflows/ci.yml:21]: Any reason we're targeting netcoreapp2.0 out of support?  Why not target 2.1

```suggestion
                version: [net452, netcoreapp2.1, netcoreapp3.1]
```

Comment (jon-armen) [reply .github/workflows/ci.yml:21]: Never mind for now, I see that the test projects are currently referencing netcoreapp2.0.  Much better to get CI up and running any make those changes in another pr.

Comment (srprash) [reply .github/workflows/ci.yml:21]: Yes. We should update our targeted frameworks to the supported ones for the whole SDK some time soon.

Comment (jon-armen) [reply .github/workflows/ci.yml:21]: I've got a PR out there to do just that.  I'll update it once we have this merged.

**Pull request merged**
